### ISSUE TITLE
should i include plugins while packaging nnn in the pkgbuild for mpr?

### ISSUE THREAD

**Issue by onlyvip** [jarun/nnn#1116]
I am trying to include nnn in the [MPR](https://mpr.hunterwittenborn.com/)
and i have a pkgbuild ready as well. i have made it so that all the plugins gets cpoied into the /opt/nnn/plugins/*, folder. so do you recommend this path and way of adding the plugins with the initial install or some other path is suitable?
here is the pkgbuild
[PKGBUILD](https://termbin.com/h2n5) 
check line 60.

**Comment by jarun** [comment]
Arch Linux packagers do that: https://archlinux.org/packages/community/x86_64/nnn/

I am not sure how they install the plugins for the user in the user-specific location.

`nnn` looks for plugins in `${XDG_CONFIG_HOME:-$HOME/.config}/nnn/plugins`.

@maximbaz may be able to help you with this.

**Comment by onlyvip** [comment]
yeah thanks i will change the plugins location to /usr/share/nnn/plugins as well then. . 
does nnn look for plugins as a global environment in case the user havent set up the plugins but the package manager has as in this case.(the package manager cannot do anything with the user space thats why).
OR can we have nnn look for another standar location within / in case the plugins arent setup in `${XDG_CONFIG_HOME:-$HOME/.config}/nnn/plugins.`?

**Comment by jarun** [comment]
> OR can we have nnn look for another standard location within / in case the plugins aren't setup in

plugins are user-specific. The path is fixed.

I am not sure how this works on Arch Linux. Sorry.

**Comment by jarun** [comment]
I would suggest you go ahead without the plugins. It's very easy to install them.

**Comment by maximbaz** [comment]
Hello, plugins that are included in Arch Linux `nnn` package are not auto-loaded (we wouldn't want to force usage of all plugins to all users anyway), they are included for two reasons mainly:

1. Offline installation: install `nnn`, board a seven hours flight without internet, decide to spend this time learning the app, while reading the manual discover that there are some awesome plugins, be happy that they are already available on your disk and you just need to symlink / copy them to the correct folder.
2. Version consistency: plugins in `/usr/share/nnn/plugins` are installed not from latest master, but from the same release as `nnn` itself. Imagine you haven't updated for a while, so your locally installed `nnn` is of version 3.7. If you install plugins from `master` branch, they may or may not work with the old nnn, however plugins that were shipped together with v3.7 were tested at the time of the release, so there will be no surprises.

To summarize, the Arch package is not trying to force usage of all plugins onto all people nor to invent a different way to install plugins, they are simply included to help in certain situations, and people are welcome to use or not use those files, and everyone is encouraged to follow project maintainer's suggestions 🙂

**Comment by jarun** [comment]
> Imagine you haven't updated for a while, so your locally installed nnn is of version 3.7. If you install plugins from master branch, they may or may not work with the old nnn

`getplugs` now checks the version of `nnn` installed and fetches the corresponding plugins (from the same release). so this is not an issue anymore.

However, I do agree there are advantages of having the plugins locally somewhere.

**Comment by onlyvip** [comment]
I decided not to include plugins and have given a link to plugin installation link in the optdepends function instead. This way the user can get plugins the way intended by arun.
[Mpr-pkgbuild](https://mpr.hunterwittenborn.com/cgit/aur.git/tree/PKGBUILD?h=nnn-git)
However after the reason provided by @maximbaz I should include the plugins installed as well just in case someone wants to try nnn while they get on their urgent fight :p
